src/github.com/stellar/gateway/bridge/handlers/request_handler_payment_test.go

```diff
@@ -312,8 +309,7 @@ func TestRequestHandlerPayment(t *testing.T) {
 				)
 				responseString := strings.TrimSpace(string(response))
 				assert.Equal(t, 400, statusCode)
-				expectedResponse := horizon.SubmitTransactionResponse{Error: horizon.PaymentMalformedAssetCode}
-				assert.Equal(t, expectedResponse.Marshal(), []byte(responseString))
+				assert.Equal(t, "{\n  \"code\": \"invalid_parameter\",\n  \"message\": \"Invalid parameter.\",\n  \"data\": {\n    \"name\": \"asset_code\"\n  }\n}", responseString)
```

Comment:
These tests are going to be really hard to maintain.  I would think about trying a couple of things to make them a little more understandable and amenable to change.

First, rather than using string comparison, I would use `assert.Equals()` to compare the parsed forms of the response and your expected result.  That way, any changes to the formatting of the response will not break the test.  Secondly, since your comparing the parsed forms, you can use the backtick string syntax to make the test more readable.

```go
  expected := MustParseJSON(`{
    "code": "invalid_parameter",
    "message": "Invalid parameter.",
    "data": {
      "name": "asset_code"
    }
  }`)

  assert.Equal(t, expected, MustParseJSON(response))
```